File: leetcode/621.TaskScheduler.js
/**
 * @param {string[]} tasks
 * @param {number} n
 * @return {number}
 */
const leastInterval = function (tasks, n) {
  const countOfTask = new Array(26).fill(0)

  let max = 0
  const A = 'A'.charCodeAt(0)

  for (const task of tasks) {
    const index = task.charCodeAt(0) - A
    countOfTask[index]++
    if (countOfTask[index] > max) {
      max = countOfTask[index]
    }
  }

  let taskCount = 0
  for (const count of countOfTask) {
    if (count === max) {
      taskCount++
    }
  }

  return Math.max((max - 1) * (n + 1) + taskCount, tasks.length)
}

console.log(leastInterval(['A', 'A', 'A', 'B', 'B', 'B'], 2))

File: leetcode/621.TaskScheduler.go
package leetcode

import "math"

func leastInterval(tasks []byte, n int) int {
	countOfTask := make([]int, 26)

	max := 0
	A := int('A')

	for _, task := range tasks {
		index := int(task) - A
		countOfTask[index]++
		if countOfTask[index] > max {
			max = countOfTask[index]
		}
	}

	taskCount := 0
	for _, count := range countOfTask {
		if count == max {
			taskCount++
		}
	}

	return int(math.Max(float64((max-1)*(n+1)+taskCount), float64(len(tasks))))
}
